xbmc/guilib/GUIBaseContainer.h

```diff
@@ -152,8 +154,8 @@ class CGUIBaseContainer : public IGUIContainer
 
   int m_pageControl;
 
-  std::vector<CGUIListItemLayout> m_layouts;
-  std::vector<CGUIListItemLayout> m_focusedLayouts;
+  std::list<CGUIListItemLayout> m_layouts;
+  std::list<CGUIListItemLayout> m_focusedLayouts;
```

Comment:
switching to list here might feel beneficial but not sure if it will be an improvement, could equally well be a degradation in performance. Hard to say without meassuring but my feeling is that it will be worse performance wise. See https://www.youtube.com/watch?v=YQs6IC-vgmo for background information